src/commands/channels/create.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2022, Nuance, Inc. and its contributors.
+ * All rights reserved.
+ *
+ * This source code is licensed under the Apache-2.0 license found in
+ * the LICENSE file in the root directory of this source tree.
+ */
+
+import {flags} from '@oclif/command'
+import chalk from 'chalk'
+import makeDebug from 'debug'
+
+import * as ChannelsAPI from '../../mix/api/channels'
+import {ChannelModalities, ChannelModality, ChannelsCreateParams} from '../../mix/api/channels-types'
+import {MixClient, MixResponse, MixResult} from '../../mix/types'
+import MixCommand from '../../utils/base/mix-command'
+import {eInvalidValue} from '../../utils/errors'
+import * as MixFlags from '../../utils/flags'
+import {DomainOption, validateChannelColor, validateChannelModeOptions} from '../../utils/validations'
+
+const debug = makeDebug('mix:commands:channels:create')
+
+export default class ChannelsCreate extends MixCommand {
+  static description = `create a new channel
+
+Use this command to create a new channel in your Mix project.
+A channel defines a collection of different modalities through
+which users interact with your application. The currently available
+modalities are:
+
+audioscript
+dtmf
+interactivity
+richtext
+tts
+
+Note that, for your convenience, this command will automatically convert
+inputs to the --mode flag as close to the format of the above modes as possible,
+by converting upper-case letters to lower-case and removing deliminating punctuation
+('-' and '_'). So, the modes 'audioscript', 'AUDIO-SCRIPT', and 'a_u-d_i-o----scriPT'
+are all equivalent to the server-side value of 'AUDIO_SCRIPT'. 
+
+The same conversion is done for colours, except that dashes ('-') are replaced 
+with ('_') rather than being deleted. So, 'light-pink', 'light____pink', and 
+'LIGHT_PINK' are also equivalent.
+
+Acceptable channel colors are:
+blue
+brown
+corn-flower
+cyan
+green
+grey
+indigo
+light-green
+light-grey
+light-orange
+light-pink
+light-purple
+orange
+pink
+purple
+ruby
+salmon
+sky
+teal
+yellow
+`
+
+  static examples = [
+    'mix channels:create -P 1922 --name "New IVR channel" --mode ivr',
+  ]
+
+  static flags = {
+    color: flags.string({
+      description: 'channel color',
+    }),
+    mode: MixFlags.modesFlag,
+    name: flags.string({
+      description: 'channel name',
+      required: true,
+    }),
+    project: MixFlags.projectFlag,
+    // output flags
+    json: MixFlags.jsonFlag,
+    yaml: MixFlags.yamlFlag,
+  }
+
+  get domainOptions(): DomainOption[] {
+    debug('get domainOptions()')
+    return ['project']
+  }
+
+  tryDomainOptionsValidation(options: any, domainOptions: DomainOption[]) {
+    debug('tryDomainOptionsValidation()')
+    super.tryDomainOptionsValidation(options, domainOptions)
+
+    if (options.color === undefined) {
+      this.warn(chalk.yellow(`The V4 API currently requires the --color flag to be set.
```

Comment:
As we are going to make `color` a required flag, we will no longer need this warning. However, validation of the color value is still required.